test/test_corpus.cc

```diff
@@ -132,6 +132,56 @@ unique_ptr<Diagnostic> errorToDiagnostic(const core::GlobalState &gs, const core
     return make_unique<Diagnostic>(Range::fromLoc(gs, error.loc), error.header);
 }
 
+class ErrorHandler {
+    vector<unique_ptr<core::Error>> &errors;
+    Expectations &test;
+    shared_ptr<core::ErrorQueue> errorQueue;
+
+public:
+    map<string, string> got;
+
+    ErrorHandler(vector<unique_ptr<core::Error>> &errors, Expectations &test, shared_ptr<core::ErrorQueue> errorQueue)
```

Comment:
Nit: You can make the `errorQueue` param a reference; I think as-is the code will cause two atomic increments of the `shared_ptr` reference counter -- one to create the argument, and one to assign to the class field.